### DISCUSSION TITLE
How should i validate a token?What is the best practice?

### DISCUSSION THREAD

**Discussion by DougTea** [keycloak/keycloak#27225] [Q&A]
As i known

The validation time of a token depends on session and and token when use token introspect endpoint.If i use this introspect endpoint,i must take care of both session and token expire time,but session expire time can only be setup by realm scope,what i need is client scope setup.And it seems odd when using oauth2 client together with session.

Another way to validate a token is jwt signature checking and jwt claim expr check.

So please tell me what is the best practice or is there any other options ?
By the way,i use keycloak to take care of the authentication of a rest api server.

**Comment by danielFesenmeyer** [comment]
Afaik, you can configure "Client Session Max" per client - in order to have this value taken into account, you should configure "SSO Session Max" (on realm level) to a value which is at least the same as your clients' maximum "Client Session Max". See https://www.keycloak.org/docs/latest/server_admin/index.html#:~:text=Session%20Idle%20configuration.-,Client%20Session%20Max,-The%20maximum%20time

And regarding token validation: If possible, it's good to use an existing, well-tested library. Here's a nice list of libraries for different languages: https://jwt.io/libraries
